usr/src/cmd/fs.d/df.c

```diff
@@ -1282,15 +1283,33 @@ adjust_total_blocks(struct df_request *dfrp, fsblkcnt64_t *total,
 		}
 
 		this_quota = _zfs_prop_get_int(zhp, ZFS_PROP_QUOTA);
-		if (this_quota && this_quota < quota)
+		if (this_quota && this_quota < quota) {
 			quota = this_quota;
+			logical = FALSE;
+		}
+
+		this_quota = _zfs_prop_get_int(zhp, ZFS_PROP_LOGICALREFQUOTA);
+		if (this_quota && this_quota < quota) {
+			quota = this_quota;
+			logical = TRUE;
+		}
+
+		this_quota = _zfs_prop_get_int(zhp, ZFS_PROP_LOGICALQUOTA);
+		if (this_quota && this_quota < quota) {
+			quota = this_quota;
+			logical = TRUE;
+		}
 
 		/* true at last iteration of loop */
 		if ((slash = strrchr(dataset, '/')) == NULL) {
 			uint64_t size;
 
-			size = _zfs_prop_get_int(zhp, ZFS_PROP_USED) +
-			    _zfs_prop_get_int(zhp, ZFS_PROP_AVAILABLE);
+			if (logical)
+				size = _zfs_prop_get_int(zhp, ZFS_PROP_LOGICALUSED) +
+					_zfs_prop_get_int(zhp, ZFS_PROP_AVAILABLE);
```

Comment:
This doesn't make sense to me.  `LOGICALUSED` and `AVAILABLE` are in different units, so I don't see how adding them together is going to end up with anything useful.

The overall logic of this loop is: quota = the smallest of: ( any quota that applies here, and the topmost dataset's (e.g. `rpool`'s) used+avail ).  I don't see why we would need to modify that based on `rpool`'s `LOGICALUSED`.